13 Audit Log Module

brianghig opened the pull request:
Beginnings of the Audit Log module. Providing several display types and base test coverage of the audit listing component.

* [x] Move Audit to its own non-admin page
* [x] Move sortObjectKeys to pipe

brianghig committed:
Initial commit of Audit Log Module import

brianghig committed:
Fixing build issues

brianghig committed:
Adding Audit entry to Admin Topics. Fixing service endpoint for retrieving audit logs

brianghig committed:
Partial commit for adding the Audit Log listing

brianghig committed:
Updating to use user match service. Displaying Actors selected in the Audit form. Properly importing stylesheet

brianghig committed:
Merge branch 'develop' into 13-Audit-Log-Module

brianghig committed:
Styling the filters consistently. Adding User audit type. Adding User Authentication audit type

brianghig committed:
Adding spec for Audit Component

brianghig committed:
Fixing lint issues with spec

tzfx commented on src/app/core/admin/audit/audit-view-change.component.ts:
```diff
@@ -0,0 +1,52 @@
+	}
+
+	// Derived from http://stackoverflow.com/a/1359808 and http://stackoverflow.com/a/23124958
+	sortObjectKeys(obj: any): any {
```
This looks like a good candidate to in a util class somewhere that's injectable and sharable across the app.

brianghig replied:
@tzfx Agreed! I think it makes sense to be its own pipe, since it's applicable to the templates. Will change.

jrassa committed:
update audit list filters to better match UX/style
moved Timestamp filter to top for better visibility
updated date range fields to use newer BsDatepicker module
update style of filter to better match quick filter component

tzfx commented on src/app/core/admin/audit/audit.component.ts:
```diff
@@ -0,0 +1,271 @@
+@Component({
+	selector: 'audit',
+	styleUrls: ['audit.scss'],
+	templateUrl: './audit-list.component.html'
```
Probably want this to be `audit.component.html` to clearly link this component and template.

jrassa replied:
I moved and renamed all corresponding files to`list-audit-entries`

jrassa committed:
fix test

jrassa committed:
rename/move AuditComponent to ListAuditEntriesComponent

jrassa commented:
Should this be under admin?  In the old starter it was its own separate page that was only accessible to users with the 'Auditor' role.  The backend is still enforcing this restriction.  When first testing I got an error since I did not have the required role.

brianghig commented:
> Should this be under admin? In the old starter it was its own separate page that was only accessible to users with the 'Auditor' role. The backend is still enforcing this restriction. When first testing I got an error since I did not have the required role.

@jrassa no, it should probably be under its own "audit" section. I'll make that change now.

brianghig committed:
Moving sortObjectKeys to pipe with test coverage

brianghig committed:
Adding test coverage for sortObjectKeys

brianghig committed:
Moving Audit Log listing to its own module with its own role protections outside of Admin

brianghig commented:
@tzfx @jrassa I believe that all comments thus far have been addressed. Thanks for the feedback!

brianghig committed:
Covering audit service with spec
